scripts/chgres_cube.py

```diff
@@ -0,0 +1,286 @@
+#!/usr/bin/env python
+"""
+The run script for chgres_cube
+"""
+
+import datetime as dt
+import logging
+import os
+import re
+import sys
+from argparse import ArgumentParser
+from copy import deepcopy
+from pathlib import Path
+
+from uwtools.api.chgres_cube import ChgresCube
+from uwtools.api.config import get_yaml_config
+from uwtools.api.fs import link as uwlink
+from uwtools.api.logging import use_uwtools_logger
+
+
+def _parse_var_defns(file):
+    var_dict = {}
+    with open(file, "r", encoding="utf-8") as f:
+        for line in f:
+            if "=" in line.strip():
+                key, value = line.split("=", 1)
+                key = key.strip()
+                value = value.strip()
+
+                if value.startswith("(") and value.endswith(")"):
+                    items = re.findall(r"\((.*?)\)", value)
+                    if items:
+                        value = [item.strip() for item in items[0].split()]
+                        var_dict[key] = value
+    return var_dict
+
+
+def _walk_key_path(config, key_path):
+    """
+    Navigate to the sub-config at the end of the path of given keys.
+    """
+    keys = []
+    pathstr = "<unknown>"
+    for key in key_path:
+        keys.append(key)
+        pathstr = " -> ".join(keys)
+        try:
+            subconfig = config[key]
+        except KeyError:
+            logging.error(f"Bad config path: {pathstr}")
+            raise
+        if not isinstance(subconfig, dict):
+            logging.error(f"Value at {pathstr} must be a dictionary")
+            sys.exit(1)
+        config = subconfig
+    return config
+
+
+def parse_args(argv):
+    """
+    Parse arguments for the script.
+    """
+    parser = ArgumentParser(
+        description="Script that runs chgres_cube via uwtools API",
+    )
+    parser.add_argument(
+        "-c",
+        "--config-file",
+        metavar="PATH",
+        required=True,
+        help="Path to experiment config file.",
+        type=Path,
+    )
+    parser.add_argument(
+        "--cycle",
+        help="The cycle in ISO8601 format (e.g. 2024-07-15T18)",
+        required=True,
+        type=dt.datetime.fromisoformat,
+    )
+    parser.add_argument(
+        "--key-path",
+        help="Dot-separated path of keys leading through the config to the driver's YAML block",
+        metavar="KEY[.KEY...]",
+        required=True,
+        type=lambda s: s.split("."),
+    )
+    parser.add_argument(
+        "--member",
+        default="000",
+        help="The 3-digit ensemble member number.",
+    )
+    return parser.parse_args(argv)
+
+
+# pylint: disable=too-many-locals, too-many-statements, too-many-branches
```

Comment:
I'm leaving this open for now to see what you think of the current shorter version. It might be a cleaner version as well if I moved the file linking or the loop off to a helper function?